ci/scripts/cpp_build.sh

```diff
@@ -283,7 +283,10 @@ else
 fi
 
 if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
-  time meson install
+    time {
+      meson compile -j $[${n_jobs} + 1];
+      meson install;
+  }
 else
   export CMAKE_BUILD_PARALLEL_LEVEL=${CMAKE_BUILD_PARALLEL_LEVEL:-$[${n_jobs} + 1]}
```

Comment:
How about defining `ARROW_BUILD_PARALLEL` or something and use it as the default for Meson and CMake?

```shell
: ${ARROW_BUILD_PARALLEL:=$[${n_jobs} + 1]}
if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
  meson compile -j ${ARROW_BUILD_PARALLEL}
else
  export CMAKE_BUILD_PARALLEL_LEVEL=${ARROW_BUILD_PARALLEL}
fi
```